lib/acmesmith/post_issueing_hooks/shell.rb

```diff
@@ -0,0 +1,23 @@
+require 'erb'
+require 'acmesmith/post_issueing_hooks/base'
+
+module Acmesmith
+  module PostIssueingHooks
+    class Shell < Base
+      class HostedZoneNotFound < StandardError; end
+      class AmbiguousHostedZones < StandardError; end
+
+      def initialize(common_name:, command:)
+        @common_name = common_name
+        @command = command
+      end
+
+      def execute()
+        parsed_command = ERB.new(@command).result(binding)
```

Comment:
I would normally agree, but the commands are configured in the config.yml. I would not know how to implement this in a user friendly way.

But I made a fallacy here. I made it possible to define a configuration per common_name and still I provide this common_name as template variable.

The use of variables make sense when the configuration looks like this:
```yaml

post_issueing_hooks:
    - shell:
        command: aws iam delete-server-certificate --server-certificate-name <%= @common_name %>-will-be-removed
```
Then the same hooks are executed for every domain.

But with a configuration per domain, there no need for variables, I could look like this:

```yaml

post_issueing_hooks:
    - shell:
      "www.example.com":
        command: aws iam delete-server-certificate --server-certificate-name www.example.com-will-be-removed
```

Second is of course more flexible, but will create endless configuration files. What do you think about this?